kani-compiler/src/codegen_cprover_gotoc/codegen/assert.rs

```diff
@@ -144,6 +145,33 @@ impl<'tcx> GotocCtx<'tcx> {
         self.codegen_cover(Expr::bool_true(), msg, span)
     }
 
+    /// Given the message for an assertion, generate a reachability check that
+    /// is meant to check whether the assertion is reachable. The function
+    /// returns a modified version of the provided message that should be used
+    /// for the assertion to allow the CBMC output parser to pair the assertion
+    /// with its reachability check.
+    /// If reachability checks are disabled, the function returns the message
+    /// unmodified and an empty (skip) statement.
+    pub fn codegen_reachability_check(
+        &mut self,
+        msg: String,
+        span: Option<Span>,
+    ) -> (String, Stmt) {
+        if self.queries.get_check_assertion_reachability() {
+            // Generate a unique ID for the assert
+            let assert_id = self.next_check_id();
+            // Generate a message for the reachability check that includes the unique ID
+            let reach_msg = GotocCtx::reachability_check_message(&assert_id);
+            // Also add the unique ID as a prefix to the assert message so that it can be
+            // easily paired with the reachability check
+            let msg = GotocCtx::add_prefix_to_msg(&msg, &assert_id);
+            // inject a reachability (cover) check
```

Comment:
I think calling this a cover is now a bit confusing. Shouldn't we start using a different property class for reachability checks and cover for actual cover? Or am I missing something?

Reply:
Good point. I'll change their property class to something else.

Reply:
Done: I introduced a new property class, `PropertyClass::ReachabilityCheck`, and switched the CBMC output parser to identify reachability checks using the property class (as opposed to the description as was done previously).

Reply:
Thanks!